dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -2685,34 +2698,169 @@ private int getJobInstanceTableVersion(PersistenceServiceUnit psu) throws Except
-                    logger.fine("The UPDATETIME column does not exist, job instance table version = 1");
-                    instanceVersion = 1;
-                    return instanceVersion;
+                    try {
+                        Integer exec = new TranRequest<Integer>(em) {
+                            @Override
+                            public Integer call() throws Exception {
+                                // Verify that UPDATETIME column exists by running a query against it.
+                                String queryString = "SELECT COUNT(x.lastUpdatedTime) FROM JobInstanceEntityV2 x";
+                                TypedQuery<Long> query = em.createQuery(queryString, Long.class);
+                                query.getSingleResult();
+                                logger.fine("The UPDATETIME column exists, job instance table version = 2");
+                                instanceVersion = 2;
+                                return instanceVersion;
+                            }
+                        }.runInNewOrExistingGlobalTran();
+
+                        return exec;
+                    } catch (javax.persistence.PersistenceException e2) {
+                        cause = e2.getCause();
+                        while (cause != null) {
+                            if (cause instanceof SQLSyntaxErrorException &&
+                                cause.getMessage() != null &&
+                                cause.getMessage().contains("UPDATETIME")) {
+                                // The column isn't there.
+                                logger.fine("The UPDATETIME column does not exist, job instance table version = 1");
+                                instanceVersion = 1;
+                                return instanceVersion;
+                            }
+                            cause = cause.getCause();
+                        }
+                    }
                 }
-                cause = cause.getCause();
             }
             logger.fine("Unexpected exception while checking job instance table version, re-throwing");
-            throw e;
+            throw e3;
         } finally {
             em.close();
         }
     }
+
+    @Override
+    public List<String> getGroupNamesForJobID(long jobInstanceID) throws NoSuchJobInstanceException {
+        EntityManager em = getPsu().createEntityManager();
+        try {
+            TypedQuery<JobInstanceEntityV3> query = em.createNamedQuery(JobInstanceEntityV3.GET_ALL_GROUPNAMES_BY_JOBINSTANCE_QUERY,
+                                                                        JobInstanceEntityV3.class);
+            query.setParameter("jobinstanceID", jobInstanceID);
+            List<JobInstanceEntityV3> rows = query.getResultList();
+
+            if (rows.isEmpty()) {
+                throw new NoSuchJobInstanceException("No job execution found for id = " + jobInstanceID);
+            }
```

Comment:
It shouldn't be an error if there are no operation groups configured, and/or no groups associated with a particular instance.   (If it is somehow I still doubt it'd be a NoSuchJobInstanceException)